functionary/core/utils/workflow.py

```diff
@@ -0,0 +1,103 @@
+from typing import TYPE_CHECKING
+
+from django.db import transaction
+
+from core.models import Workflow, WorkflowStep
+
+if TYPE_CHECKING:
+    from core.models import Function
+
+
+def add_step(
+    workflow: Workflow,
+    name: str,
+    function: "Function",
+    parameter_template: str,
+    next: WorkflowStep | None = None,
+) -> WorkflowStep:
+    """Add a WorkflowStep to the specified point in the Workflow
+
+    Args:
+        workflow: Workflow to add a step to
+        name: Name of the WorkflowStep
+        function: Function to task
+        parameter_template: Template string that will be rendered to form the
+            parameter json for the function
+        next: The WorkflowStep to insert the new step before. The default value
+            of None will insert at the end of the Workflow.
+
+    Returns:
+        The created WorkflowStep
+
+    Raises:
+        ValueError: workflow and next.workflow do not match
+    """
+    if next is not None and workflow != next.workflow:
+        raise ValueError("Provided next step is not part of provided workflow")
+
+    previous_step = WorkflowStep.objects.filter(workflow=workflow, next=next).first()
```

Comment:
Nit: can you make these variable names consistent across the functions, this one is `previous` and the other two are `before` and `before_step` steps.

Reply:
Changed to `before_step` to match the others.